Submitted By: Jim Gifford (jim at linuxfromscratch dot org)
Date: 2004-01-30
Initial Package Version: 2.6
Origin: Gentoo
Description: Fixes Compile Issues with the 2.6 Kernel
 
diff -Naur net-tools-1.60.orig/lib/x25_sr.c net-tools-1.60/lib/x25_sr.c
--- net-tools-1.60.orig/lib/x25_sr.c	2000-05-20 13:38:10.000000000 +0000
+++ net-tools-1.60/lib/x25_sr.c	2004-01-30 19:46:39.000000000 +0000
@@ -22,6 +22,7 @@
 #include <sys/socket.h>
 #include <sys/ioctl.h>
 #include <linux/x25.h>
+#include <linux/version.h>
 #include <ctype.h>
 #include <errno.h>
 #include <netdb.h>
@@ -77,7 +78,11 @@
   rt.sigdigits=sigdigits;
 
   /* x25_route_struct.address isn't type struct sockaddr_x25, Why? */
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 0)
   memcpy(&rt.address, &sx25.sx25_addr, sizeof(x25_address));
+#else
+  memcpy(&rt.address, &sx25.sx25_addr, sizeof(struct x25_address));
+#endif
 
   while (*args) {
 	if (!strcmp(*args,"device") || !strcmp(*args,"dev")) {
